lib/IDE/Refactoring.cpp

```diff
@@ -2070,6 +2070,211 @@ bool RefactoringActionExpandTernaryExpr::performChange() {
   return false; //don't abort
 }
 
+bool RefactoringActionConvertIfLetExprToGuardExpr::
+	isApplicable(ResolvedRangeInfo Info, DiagnosticEngine &Diag) {
+
+  if (Info.Kind != RangeKind::SingleStatement
+      && Info.Kind != RangeKind::MultiStatement)
+    return false;
+
+  if (Info.ContainedNodes.empty())
+    return false;
+
+  IfStmt *If = nullptr;
+
+  if (Info.ContainedNodes.size() == 1) {
+    if (auto S = Info.ContainedNodes[0].dyn_cast<Stmt*>()) {
+        If = dyn_cast<IfStmt>(S);
+    }
+  }
+
+  if (!If)
+    return false;
+
+  auto CondList = If->getCond();
+
+  if (CondList.size() == 1) {
+      auto E = CondList[0];
+      auto P = E.getKind();
+      if (P == swift::StmtConditionElement::CK_PatternBinding)
+        return true;
+  }
+
+  return false;
+}
+	
+bool RefactoringActionConvertIfLetExprToGuardExpr::performChange() {
+
+  IfStmt *If = nullptr;
+
+  if (RangeInfo.ContainedNodes.size() == 1) {
+    if (auto S = RangeInfo.ContainedNodes[0].dyn_cast<Stmt*>()) {
+        If = dyn_cast<IfStmt>(S);
+    }
+  }
+
+  if (!If)
+    return true; // abort
+
+  auto CondList = If->getCond();
+
+  if (CondList.size() != 1)
+    return true; // abort
+
+  auto E = CondList[0];
+  auto P = E.getPatternOrNull();
+  if (!P)
+    return true; // abort
+
+    // Get if-let condition
+    SourceRange range = E.getSourceRange();
+    SourceManager &SM = RangeInfo.RangeContext->getASTContext().SourceMgr;
+    auto CondCharRange = Lexer::getCharSourceRangeFromSourceRange(SM, range);
+
+    auto Body = dyn_cast_or_null<BraceStmt>(If->getThenStmt());
+
+    if (!Body)
+        return true; // abort
```

Comment:
Could you move the `ThenStmt` check to the `isApplicable` function too? so we don't need to check it here and surprising the users by aborting.

Reply:
Moved to the `isApplicable`.